core/src/test/java/org/apache/iceberg/TestOverwrite.java

```diff
@@ -295,5 +343,6 @@ public void testValidatedOverwriteWithAppendSuccess() {
         .hasMessageStartingWith("Cannot append file with rows that do not match filter");
 
     assertThat(latestSnapshot(base, branch).snapshotId()).isEqualTo(baseId);
+    assertThat(latestSnapshot(table, branch).operation()).isEqualTo(DataOperations.APPEND);
```

Comment:
Shouldn't this case be an overwrite? Since we'd be removing files which match the filter in `overwriteByRowFilter`?

Reply:
Oh I see the overwrite fails, is this assertion just for verifying what the operation is for the table that's setup for the tests (the append that's done in the BeforeEach)? If so I'd maybe just leave that assertion out (and we can make sure the tests for AppendFiles properly verifies the operation).

Reply:
you're right, it's a bit confusing and probably better to leave this assertion out